apps/web/pages/api/auth/saml/userinfo.ts

```diff
@@ -24,7 +24,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     // check for query param
     if (!token) {
       let arr: string[] = [];
-      arr = arr.concat(req.query.access_token);
+      arr = arr.concat(req.query.access_token!);
```

Comment:
Let's avoid null assertion, instead validate `req.query`:

Whenever we are parsing unknown json where we know the shape the object should have but it is not typed from source, it is worth the extra effort to use zod so we get both type and runtime safety.

📄 [From Cal.com's Gitbook](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/validate-using-zod-instead-of-type-casting)

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix